src/mcmc/gibbs.jl

```diff
@@ -478,11 +547,54 @@ function setparams_varinfo!!(
 end
 
 function setparams_varinfo!!(
-    model::DynamicPPL.Model, sampler::Sampler, state::PGState, params::AbstractVarInfo
+    model::DynamicPPL.Model, sampler::Sampler{<:PG}, state::PGState, params::AbstractVarInfo
 )
     return PGState(params, state.rng)
 end
 
+"""
+    match_linking!!(varinfo_local, prev_state_local, model)
+
+Make sure the linked/invlinked status of varinfo_local matches that of the previous
+state for this sampler. This is relevant when multilple samplers are sampling the same
+variables, and one might need it to be linked while the other doesn't.
+"""
+function match_linking!!(varinfo_local, prev_state_local, model)
+    prev_varinfo_local = varinfo(prev_state_local)
+    was_linked = DynamicPPL.istrans(prev_varinfo_local)
+    is_linked = DynamicPPL.istrans(varinfo_local)
+    if was_linked && !is_linked
+        varinfo_local = DynamicPPL.link!!(varinfo_local, model)
+    elseif !was_linked && is_linked
+        varinfo_local = DynamicPPL.invlink!!(varinfo_local, model)
+    end
+    # TODO(mhauru) The above might run into trouble if some variables are linked and others
```

Comment:
We should probably have an issue accompany this 👍

Reply:
Opened here: https://github.com/TuringLang/Turing.jl/issues/2401